pontoon/base/models/user.py

```diff
@@ -234,19 +235,55 @@ def badges_translation_count(self):
 @property
 def badges_review_count(self):
     """Translation reviews provided by user that count towards their badges."""
+    # Exclude auto-rejections caused by creating a new translation or approving an existing one
+    closely_preceded_action = ActionLog.objects.filter(
+        performed_by=OuterRef("performed_by"),
+        action_type__in=["translation:created", "translation:approved"],
+        created_at__gt=OuterRef("created_at"),
+        created_at__lte=OuterRef("created_at") + timedelta(milliseconds=100),
+    )
+
     return self.actions.filter(
-        Q(action_type="translation:approved") | Q(action_type="translation:rejected"),
+        Q(action_type="translation:approved")
+        | Q(
+            ~Exists(closely_preceded_action),
+            action_type="translation:rejected",
+        ),
```

Comment:
This code should be reverted.

Reply:
I meant this whole change introduced in https://github.com/mozilla/pontoon/pull/3442/commits/2b696be27e371766a0aeb92f02f945c198989cac should be reverted, because it turned out not to scale well. That's why we are making the current changes in favour of potentially 2 badges awarded at the same time.